rust/kernel/file.rs

```diff
@@ -8,6 +8,104 @@
 use crate::{bindings, cred::CredentialRef, error::Error, Result};
 use core::{mem::ManuallyDrop, ops::Deref};
 
+/// Access mode associated with a file
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum AccessMode {
+    /// File is read only.
+    ReadOnly,
+    /// File is write only.
+    WriteOnly,
+    /// File is readable and writeable.
+    ReadWrite,
+}
+
+/// Flags associated with a [`File`].
+///
+/// Wrapper around `struct file`'s `f_flags`.
+#[repr(transparent)]
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub struct FileFlags(u32);
+
+impl FileFlags {
+    /// Also known as `O_APPEND`.
+    pub const APPEND: u32 = bindings::O_APPEND;
+
+    /// Also known as `FASYNC`.
+    pub const ASYNC: u32 = bindings::FASYNC;
+
+    /// Also known as `O_CLOEXEC`.
+    pub const CLOSE_ON_EXEC: u32 = bindings::O_CLOEXEC;
```

Comment:
I appreciate the better names, but I am not sure about changing them... These names are very well-known (some coming from standards) and, in general, we are trying to keep naming close to the C side (at least for the moment).

Reply:
Sure, I was on the fence about this anyways.

Should we do with or without `O_` prefix?

Reply:
I think removing the prefix would be the "worst of both worlds", i.e. a middle ground where we are not using the same names but we are not improving the names either.